skimage/feature/_hog.py

```diff
@@ -154,14 +188,26 @@ def hog(image, orientations=9, pixels_per_cell=(8, 8), cells_per_block=(3, 3),
         # to avoid problems with subtracting unsigned numbers
         image = image.astype('float')
 
-    gx = np.empty(image.shape, dtype=np.double)
-    gx[:, 0] = 0
-    gx[:, -1] = 0
-    gx[:, 1:-1] = image[:, 2:] - image[:, :-2]
-    gy = np.empty(image.shape, dtype=np.double)
-    gy[0, :] = 0
-    gy[-1, :] = 0
-    gy[1:-1, :] = image[2:, :] - image[:-2, :]
+    if multichannel:
+        g_row_by_ch = np.empty_like(image, dtype=np.double)
+        g_col_by_ch = np.empty_like(image, dtype=np.double)
+        g_magn = np.empty_like(image, dtype=np.double)
+
+        for idx_ch in range(image.shape[2]):
+            g_row_by_ch[:, :, idx_ch], g_col_by_ch[:, :, idx_ch] = \
+                _hog_channel_gradient(image[:, :, idx_ch])
+            g_magn[:, :, idx_ch] = np.hypot(g_row_by_ch[:, :, idx_ch],
+                                            g_col_by_ch[:, :, idx_ch])
+
+        # For each pixel select the channel with the highest gradient magnitude
+        idcs_max = g_magn.argmax(axis=2)
+        rr, cc = np.meshgrid(np.arange(image.shape[0]),
+                             np.arange(image.shape[1]),
+                             indexing='ij')
```

Comment:
I think you can also set `sparse=True` in `meshgrid` to save memory.  The indexing below should still work by broadcasting.

Reply:
Thanks, done!